backend/src/common/repository.py

```diff
@@ -39,6 +40,33 @@ def create(cls, session: Session, instance: T) -> T:
         session.refresh(instance)
         return instance
 
+    @classmethod
+    def create_all(cls, session: Session, instances: list[T]) -> list[T]:
+        session.add_all(instances)
+        session.commit()
+        for instance in instances:
+            session.refresh(instance)
+        return instances
+
+    @classmethod
+    def update(cls, session: Session, instance: T, update_data: dict) -> T:
+        try:
+            db_instance = (
+                session.query(cls.__model__)
+                .where(cls.__model__.id == instance.id)
+                .one()
+            )
+        except NoResultFound:
+            raise ValueError(
+                f"{cls.__model__.__name__} with id={instance.id} not found"
+            )
+        else:
```

Comment:
I think this else is not needed

Reply:
Why?

Reply:
Else runs if there is no exception, but we return from the function on line 60 anyways if there is exception so we can be sure that if there is any exception (NoResults we just handle) anything under else will run